src/object-store/src/object_accessor.hpp

```diff
@@ -286,31 +286,32 @@ namespace realm {
             throw MutationOutsideTransactionException("Can only create objects within a transaction.");
         }
 
-        // get or create our accessor
-        bool created;
-
         // try to get existing row if updating
         size_t row_index = realm::not_found;
         realm::TableRef table = ObjectStore::table_for_object_type(realm->read_group(), object_schema.name);
         const Property *primary_prop = object_schema.primary_key_property();
+
         if (primary_prop) {
             // search for existing object based on primary key type
             ValueType primary_value = Accessor::dict_value_for_key(ctx, value, object_schema.primary_key);
-            if (primary_prop->type == PropertyTypeString) {
-                row_index = table->find_first_string(primary_prop->table_column, Accessor::to_string(ctx, primary_value));
-            }
-            else {
-                row_index = table->find_first_int(primary_prop->table_column, Accessor::to_long(ctx, primary_value));
-            }
 
-            if (!try_update && row_index != realm::not_found) {
-                throw DuplicatePrimaryKeyValueException(object_schema.name, *primary_prop,
-                    "Attempting to create an object of type '" + object_schema.name + "' with an exising primary key value.");
+            if (!Accessor::is_null(ctx, primary_value)) {
```

Comment:
If `primaryKey` is null or undefined here, then that is caught later in the function if it has no default value.